### ISSUE TITLE
make it possible to configure the stability level via a CLI command

### ISSUE THREAD

**Issue by lsmith77** [composer/composer#1411]
use case is for example a travis-ci setup where one wants to test both the current stable version and the current development version. however the development version may have indirect development requirements.

see for example https://travis-ci.org/symfony-cmf/CoreBundle/builds/3607955

i guess something like the following would be good

```
env:
  - SYMFONY_VERSION=2.1.*,STABILITY=stable
  - SYMFONY_VERSION=dev-master,STABILITY=dev

before_script:
  - composer require symfony/framework-bundle:${SYMFONY_VERSION} -stability=${STABILITY}
  - composer install --dev
```

Or a separate command:

```
  - composer set stability ${STABILITY}
```

**Comment by SplicePHP** [comment]
I know this is a bit dirty but this is the method I am currently using for my installer script. I have a bash script that creates a cakephp project and then adds requirements to the project. Since I am in development phase I need stability to be set to dev in order to get all the latest master changes in all my repositories. Because composer does not seem to have the ability to do this I had to resort to using 'sed' to update the stability setting. If I had to update the json file manually it would cause me to have to break out of the install process, edit the composer.json file and then run a second bash script to complete the installation. This would be a retarded way of doing things.

For those of you facing the same frustration, here is a single line that takes care of this.

sed -i '' 's/"minimum-stability": "stable"/"minimum-stability": "dev"/g' composer.json

If there is a better way of doing this, I would be glad to hear about it.

**Comment by curry684** [comment]
I wouldn't use bash for this to begin with - you can also use PHP as shell script, and PHP can actually handle and create nested data and JSON, if need be from the bash file. So just create a file called `create-project` like this:
```
#!/usr/bin/env php
<?php

$config = [
  "require" => [
    "foo/bar" => "^6.8.4",
  ],
  "minimum-stability" => "dev",
];

file_put_contents(__DIR__ . '/composer.json', json_encode($config, JSON_PRETTY_PRINT));
```

**Comment by staabm** [comment]
@SplicePHP 

> If there is a better way of doing this, I would be glad to hear about 

`composer config minimum-stability ${STABILITY}`

**Comment by SplicePHP** [comment]
Thanks for the suggestions.

@staabm 
This is the first thing I tried.
This is the error I get:
 [InvalidArgumentException]                                                   
  Setting minimum-stability does not exist or is not supported by this comman  
  d

**Comment by curry684** [comment]
If you implement my suggestion please note that I just edited a nasty bug out of it hehe.

**Comment by Seldaek** [comment]
Fixed by 135783299af0281db918c103cceb2b202ae154f2
